List groupings and groups in a classroom

dinever opened the pull request:
This is part of the team management interface #292. To make sure a PR doesn't get too large to view, I'm breaking the team management feature into smaller PRs.

I created two endpoints: `organizations#show_groupings` and `groupings#show`.

![demo](https://cloud.githubusercontent.com/assets/1311594/16178569/423b7caa-361a-11e6-97d9-78a732bb1501.gif)

Showcase:

![screen shot 2016-06-19 at 12 34 31 pm](https://cloud.githubusercontent.com/assets/1311594/16178582/9e660aa4-361a-11e6-9c71-a802ae3ee904.png)

![screen shot 2016-06-19 at 12 34 40 pm](https://cloud.githubusercontent.com/assets/1311594/16178584/a0d35954-361a-11e6-8fac-25a609a25dfb.png)

![screen shot 2016-06-19 at 12 34 48 pm](https://cloud.githubusercontent.com/assets/1311594/16178585/a3dad212-361a-11e6-8843-77c337603d24.png)

Changes are hidden behind the Flipper `team_management`. Would ❤️  to hear your feedback!

dinever commented on app/controllers/groupings_controller.rb:
```diff
@@ -0,0 +1,16 @@
+  private
+
+  def set_grouping
+    @grouping = Grouping.find_by!(id: params[:id])
```
Should use slug here. I'm going to create a migration and make `grouping` sluggable.

tarebyte commented on app/views/groupings/_group.html.erb:
```diff
@@ -0,0 +1,23 @@
+        <% end %>
+      <% end %>
+      <% if group.repo_accesses.count > 7 %>
+        <%= link_to '#' do %>
```
I think you want `group.github_team.html_url` here

dinever replied:
@tarebyte Actually I wanted this link to lead the teachers to a `group#show` route, which hasn't been implemented yet.

However, since we will implement a drag-and-drop feature on this page. It will need some UI adjustment so that we can show not only 7 but all students in each group.

dinever commented on app/views/groupings/show.html.erb:
```diff
@@ -0,0 +1,35 @@
+                    <%= render partial: "group", locals: { group: group } %>
+                <% end %>
+            <% end %>
+          </div>
```
Hi @jasonlong, we're implementing a team management interface here. Teachers will be able to move students from a team to another by simply drag and drop the student avatars. 

It would be great if you can give me some feedback on this UI. Thanks!

/cc @johndbritton @tarebyte

jasonlong replied:
Visually, I think the Masonry-style layout is nice. A couple thoughts though:

1. There doesn't seem to be any sort of affordance for users to know that these are drag and droppable. If nothing else, there should probably be a one-liner at the top of the page saying so.

2. If you're dragging an avatar to a drop area that's above or below the current viewport, that operation will probably be impractical. Will there be a fallback method?

3. The site uses a gray heading and white content area for the various boxes and these boxes flip that to have a gray content area.

4. Minor design nit: the line with the student count should be tightened up to reduce the whitespace above it.

dinever commented:
Update:

- [x] Add a `flash` to remind users that the student avatars are drag-and-droppable.
- [x] Change the style to gray heading and white content area.
- [x] Reduce the line height above the student count.

Before:

![screen shot 2016-07-14 at 10 53 19 pm](https://cloud.githubusercontent.com/assets/1311594/16843949/cb64b76c-4a15-11e6-8b86-da9a43d16f8f.png)

After:

![screen shot 2016-07-14 at 10 41 08 pm](https://cloud.githubusercontent.com/assets/1311594/16843950/cc8d852e-4a15-11e6-8d23-f2fb06fff3a6.png)

/cc @tarebyte @jasonlong

jasonlong commented:
My 2 cents:

* Could we fix the team cards so they align with the left and right page margins?
* I'm not sure the message at the top needs to be in flash container. Could we try just having it show as regular text. Maybe the **drag and drop** phrase could be bold to make it stand out?
* The student count still feels like it makes the heading unnecessarily tall to me. Could that text (if it's really needed) be in a lighter gray next to the team name?

dinever commented:
@jasonlong Thanks a lot for the feedback! 😄 

I made some changes to the style.

Before:

![](https://cloud.githubusercontent.com/assets/1311594/16843950/cc8d852e-4a15-11e6-8d23-f2fb06fff3a6.png)

After:

![screen shot 2016-07-15 at 12 52 23 am](https://cloud.githubusercontent.com/assets/1311594/16848085/b1d358b0-4a26-11e6-8123-5296f62f5e7d.png)

BTW maybe we can also add a ![screen shot 2016-07-15 at 1 00 10 am](https://cloud.githubusercontent.com/assets/1311594/16848260/859a3d30-4a27-11e6-8278-58950faaed13.png) before the message(like dotcom showing ProTip messages at the bottom of each PR). What do you think? 😄 

![screen shot 2016-07-15 at 1 01 43 am](https://cloud.githubusercontent.com/assets/1311594/16848362/08295b0a-4a28-11e6-8cd0-9f15d13bab09.png)

/cc @tarebyte

jasonlong commented:
Awww yeah, that's feeling a lot better to me. The **ProTip** thing could be a nice touch. I think if we did that though, that blurb should go at the bottom of the page because it's like a "oh, did you know this?" vs. "you need to know this". If dragging and dropping will be a core part of the use of this page, I'd probably keep it at the top minus the ProTip label.

Oh, one last super-duper minor nit: we don't normally use italics, so I think just the lighter color would be sufficient for the student count.

dinever commented:
@jasonlong Thanks a lot for the feedback! 😄  I dropped the `ProTip` thing since drag-n-drop will be a core feature of this page.

Update:

- [x] Removed the italic font style.

![screen shot 2016-07-15 at 11 17 41 am](https://cloud.githubusercontent.com/assets/1311594/16862638/cd2b361c-4a7d-11e6-9a31-6cfb83cae746.png)

@tarebyte I wonder if you can take another 👀  at this. After this one being merged we can move on to another PR. Thanks!

tarebyte commented on app/controllers/application_controller.rb:
```diff
@@ -22,6 +22,10 @@ def peek_enabled?
 
   private
 
+  def ensure_team_management_flipper_is_enabled?
```
Can we cut the `?` out of this method?

dinever replied:
Nice catch! 😅  Done in 716c7063484b58ce7199a7a988895626288f1a32.

tarebyte commented on spec/controllers/groupings_controller_spec.rb:
```diff
@@ -0,0 +1,24 @@
+      expect(response.status).to eq(200)
+      expect(assigns(:grouping)).to_not be_nil
+    end
+  end
```
Just for sanity sake let's add a test here to show if the user doesn't have the flag.

dinever replied:
Done in 98fa77e8976c66fca1315048f61c0b3d0df8b150. Thanks! 👍

tarebyte commented:
@dinever I'm seeing double when I create a new Group.

![screen shot 2016-07-16 at 14 15 02](https://cloud.githubusercontent.com/assets/564113/16896382/c0852228-4b5f-11e6-96a1-751b142c8b47.png)

dinever commented:
@tarebyte Thanks for pointing that out. 👍  I've fixed that in 3dc2503e52e002b1e8b237de608026c61652e492.

tarebyte commented:
👍 

![screen shot 2016-07-16 at 17 39 28](https://cloud.githubusercontent.com/assets/564113/16897336/4b7ef73e-4b7c-11e6-9361-b811ccd477c3.png)
